crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -359,6 +359,27 @@ impl Analyzer<'_, '_> {
                             types.dedup_type();
                             types.retain(|ty| !ty.is_never());
 
+                            if types.iter().any(|t| t.is_any()) {
+                                return Ok(Cow::Owned(Type::any(ty.span, Default::default())));
+                            } else if types.iter().any(|t| t.is_unknown()) {
+                                return Ok(Cow::Owned(Type::unknown(ty.span, Default::default())));
+                            } else if types.iter().any(|t| {
+                                t.is_kwd(TsKeywordTypeKind::TsNumberKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsBooleanKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsStringKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsBigIntKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsSymbolKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsObjectKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsVoidKeyword)
+                                    || t.is_kwd(TsKeywordTypeKind::TsIntrinsicKeyword)
+                                    || t.is_type_lit()
+                                    || t.is_instance()
+                                    || t.is_interface()
+                            }) {
+                                if !self.rule().strict_null_checks {
```

Comment:
This is too dangerous. 
There are some cases where we have to preserve the list of types in a union or an intersection

Reply:
I think you can use the preserve option.

Reply:
This behavior should be opt-in

Reply:
I think too, But I've never seen such a situation. Is there an example of ts that fits the following situation?

Reply:
There was, but I didn't write it somewhere and I don't remember

Reply:
I don't understand your opinion

`null` and `undefined` is right that A and B disappear if they are not strict.

I can't think of anything, and if it's a special case,
These are similar cases to TypeParam and Condition.
If there's a problem then,
Someone can add a condition to this part.

If not, please make me understand through ts example.